### DISCUSSION TITLE
Helpful answers to unhelpful complaint

### DISCUSSION THREAD

**Discussion by QSDDS** [robhagemans/pcbasic#177] [General]
Sorry but so much problems, i can't do anything
- no **poke** **peek** and **call**
- **sound** with only 2 parameters
- **put** and **get** not working with **screen** 9 (640x400 16 colors)
- can't read arrow keys : a$=inkey$   print asc(a$)
and others and others
With pc-basic 2.0.4 on raspberry pi 3B+ and raspbian

**Comment by robhagemans** [comment]
If something does not work, please file a bug report specifying what you did, what you expected to happen and what happened instead.
As it stands your message is "unusable" itself.

One example, sound *does* have four parameters _in Tandy or PCjr mode_, while GWBASIC's SOUND only has 2 parameters, so if you need those parameters you need to choose tandy/pcjr syntax See the documentation https://robhagemans.github.io/pcbasic/doc/2.0/#SOUND.

What is "not working" for `PUT` and `GET` in `SCREEN 9`? You are not giving any information.
What are you trying to do with arrow keys?

`PEEK` and `POKE` have very partial implementations as we're not emulating the whole PC, if that's what you need you need to look at a PC emulator and run GW-BASIC on it.

"And others and others" - what others? If something does not work, please report it, just saying "lots of things don't work" is extremely unhelpful.

**Comment by Marrin** [comment]
@QSDDS The limitations with `PEEK`, `POKE`, and `CALL` are documented.  Also the number of arguments of `SOUND` and the condition under which there are four of them.

I suspect you can *read* the arrow keys just fine and the `ASC` is the source of an illegal function call — that's because `INKEY$` does not wait for a key to be pressed but returns with an empty string when there is no key to process, and `ASC` obviously has no way to turn nothing into a number.  Again all documented.

Just in case your problem with `GET`, `PUT`, and `SCREEN 9` is missing pixel lines at the bottom of the screen: mode 9 is 640×350 and not 640×400.  Also as documented.

This small test programs works for me:

```basic
10 SL=32:SCREEN 9:CLS:FOR I=0 TO 15:CIRCLE (SL\2-I*0,SL\2-I*0),I,I,,,1:NEXT
20 DIM A%(SL^2\2+2):GET (0,0)-(SL,SL),A%:CLS
30 FOR I=0 TO 4:FOR J=0 TO 4:PUT (I*SL,J*SL),A%:NEXT:NEXT
```
